src/compiler/types.ts

```diff
@@ -7350,7 +7366,7 @@ export const enum ScriptKind {
 // NOTE: We must reevaluate the target for upcoming features when each successive TC39 edition is ratified in
 //       June of each year. This includes changes to `LanguageFeatureMinimumTarget`, `ScriptTarget`,
 //       transformers/esnext.ts, commandLineParser.ts, and the contents of each lib/esnext.*.d.ts file.
-export const enum ScriptTarget {
+export enum ScriptTarget {

```

Comment:
This is unrelated and should be reverted; the const enum here is a perf optimization when using tsc emit.

Reply:
This is required to get the name of the target for ts1501. Perhaps there’s a better way to do so?

Reply:
Yeah, you definitely do not want to use the reverse mapping for anything except debugging in the TS repo. I'm not sure what we typically use to do this sort of message. I think we just hardcode them in the diagnostics but that doesn't sound fun. But we shouldn't use the reverse mapping.

Reply:
Per @sheetalkamat, it looks like the thing to do is:

```ts
forEachEntry(targetOptionDeclaration.type, (value, key) => value === getEmitScriptTarget(options) ? key : undefined)
```

It would be nice to pull that code out into a utility function as we have copied that in at least two places.

Reply:
I’ve done the refactor for you. (Better use reverse mapping though, I think)

Reply:
But then all the target names have become lowercase and `ES2015` is now `es6`. Are they desirable?

Reply:
> I’ve done the refactor for you. (Better use reverse mapping though, I think)

The reverse mapping isn't guaranteed to be stable if two enum values end up with the same value, but maybe that's alright. In any case, we just can't turn this into a non-const enum without consequence. It may be possible to extract `formatEnum` out of `ts.Debug`, but that would require other changes and I think can come later.

> But then all the target names have become lowercase and ES2015 is now es6. Are they desirable?

I don't feel strongly, but @DanielRosenwasser might. This does theoretically match other diagnostics (e.g. see the `Parse empty options of --target.js` baseline). But, we're definitely inconsistent.

The `es6` thing is the worst bit, and that comes down to the order of the array. We don't ever refer to to it as `es6` in any diagnostic, only `es2015`, so that's not great. Perhaps we need to iterate the list in reverse. Or, just hardcode a switch case like `getDefaultLibFileName`.

Reply:
How about just swapping the order and changing the case in `targetOptionDeclaration`, and add `.toLowerCase()` to other usages as appropriate?

Reply:
Oh sorry, what I just meant to say was the `makeReverseMap` method in the scanner (I just realized this is a local function, perhaps we should move it to utilities)

Reply:
If by `makeReverseMap`, you mean that something that reads the enum object itself to build the mapping, I wouldn't want to do that. We have `ts.Debug.formatEnum` which does that, but we don't use it for anything other than debugging or debug failures because it's not guaranteed to be human readable or consistent, just "good enough" for us to read when debugging or viewing stack traces.

Reply:
Ah, I mean that on `targetOptionDeclaration.type`